src/testdir/test_vim9_class.vim

```diff
@@ -1180,6 +1180,36 @@ def Test_class_member()
   END
   v9.CheckScriptFailure(lines, 'E1010:')
 
+  # Test for setting a member on a null object
+  lines =<< trim END
+    vim9script
+    class A
+        this.val: string
+    endclass
+
+    def F()
+        var obj: A
+        obj.val = ""
+    enddef
+    F()
+  END
+  v9.CheckScriptFailure(lines, 'E1360: Using a null object')
+
+  # Test for accessing a member on a null object
+  lines =<< trim END
+    vim9script
+    class A
+        this.val: string
+    endclass
+
+    def F()
+        var obj: A
+        echo obj.val
+    enddef
+    F()
+  END
+  v9.CheckScriptFailure(lines, 'E1360: Using a null object')
```

Comment:
Can you also add a test for accessing (read/write) the uninitialized object from the script level?

Reply:
Ok

Reply:
Added, but note that at script level it wasn't crashing. It gives a wrong error message in the case the variable is set and not accessed, but it's not a big deal.

Reply:
> Added, but note that at script level it wasn't crashing. It gives a wrong error message in the case the variable is set and not accessed, but it's not a big deal.

Actually, maybe it is.
@mg979 @yegappan  I just updated. What's the correct error? I can incorporate a fix into a PR I'm opening soon.

```
make test_vim9_class
...
Executed 53 tests                        in   0.824146 seconds
1 FAILED:
Found errors in Test_class_member():
command line..script /src/tools/vim/src/testdir/runtest.vim[601]..function RunTheTest[54]
    ..Test_class_member[170]..<SNR>9_CheckScriptFailure line 6: ['vim9script', 'class A', '
    this.val: string', 'endclass', '', 'def F()', '    var obj: A', '    obj.val = ""', 'enddef',
    'F()']: Expected 'E1360: Using a null object' but got 'E46: Cannot change read-only variable
    "val"': ['vim9script', 'class A', '    this.val: string', 'endclass', '', 'def F()', '    var
    obj: A', '    obj.val = ""', 'enddef', 'F()']

make: *** [Makefile:70: test_vim9_class] Error 1
```

Reply:
> > Added, but note that at script level it wasn't crashing. It gives a wrong error message in the case the variable is set and not accessed, but it's not a big deal.
> 
> Actually, maybe it is. @mg979 @yegappan I just updated. What's the correct error? I can incorporate a fix into a PR I'm opening soon.

@yegappan Oops, I just read the message more carefully. It might be a bug in 
```
patch 9.0.1829: Vim9 missing access-checks for private vars
```

Reply:
@errael tests must be fixed because they were based on master where `public` didn't work.